src/modules/mavlink/mavlink_main.cpp

```diff
@@ -2341,6 +2350,26 @@ Mavlink::task_main(int argc, char *argv[])
 			}
 		}
 
+		check_requested_subscriptions();
+
+		/* update streams */
+		for (const auto &stream : _streams) {
+			stream->update(t);
```

Comment:
Great work!

I guess the main timesync latency change comes from this being above the ulog streaming, and a lesser amount the stream ordering? Maybe updating the `t`  here to the latest value would help too?

Reply:
Initially I thought so too, but the streams reordering made a noticeable difference as well.
I hadn't checked, but I expect that loop to be quick. What matters (most) is how much data there is already in the TX buffer.